app/logical/tags_preview.rb

```diff
@@ -0,0 +1,41 @@
+class TagsPreview
+  def initialize(tags: nil)
+    @tags = Tag.scan_tags(tags).map {|x| {a: x, type: 'tag'}}
+    aliases
+    implications
+    tag_types
+  end
+
+  def aliases
+    names = @tags.map{ |tag| tag[:a] }.reject {|y| y.blank?}
+    aliased = TagAlias.to_aliased_with_originals(names).reject {|k,v| k == v }
+    @tags.map! do |tag|
+      if aliased[tag[:a]]
+        {a: tag[:a], b: aliased[tag[:a]], type: 'alias'}
+      else
+        tag
+      end
+    end
+  end
+
+  def implications
+    names = @tags.map {|tag| tag[:b] || tag[:a] }
+    implications = TagImplication.descendants_with_originals(names)
+    implications.each do |imp|
+      @tags += imp[1].map { |x| {a: imp[0], b: x, type: 'implication'} }

```

Comment:
recommend `<<` instead of `+=` here

Reply:
<< does not work here, it creates the wrong structure, as I'm trying to add the items from the created array to another array, not append the array itself to the other array.